client/engine/store/store.go

```diff
@@ -0,0 +1,16 @@
+// Package Store contains the interface for a go-nitro store.
+package store // import "github.com/statechannels/go-nitro/client/engine/store"
+
+import (
+	"github.com/statechannels/go-nitro/protocols"
+	"github.com/statechannels/go-nitro/types"
+)
+
+// Store is responsible for persisting objectives, objective metadata, states, signatures, private keys and blockchain data
+type Store interface {
+	GetObjectiveById(protocols.ObjectiveId) protocols.Objective // Read an existing objective
+	GetObjectiveByChannelId(types.Bytes32) protocols.Objective  // Get the objective that currently owns the channel with the supplied ChannelId
```

Comment:
I think if the client was entirely objective driven, then perhaps one would suffice. But it feels to me as if we may not be able to make it entirely objective driven -- it may need to operate in a "channel-centric" at times.

There's an example of each store method being used on this PR. 

The first, when a counterparty has sent us signatures for a particular objective which we are both running. Having one objective covering multiple channels seems nice in that situation, since things like virtual funding involve multiple channels.

The second, when our chain service spots that a challenge has been raised for a channel we participate in. In this case, the channel may be owned by (say) a `direct-funder` or `virtual-funder` objective. The chain service wouldn't know which, but the store would know and would return it via the second method. Both objectives would have a handler for this event.

Reply:
Is it correct that a given channel can have a single objective at any given time?